### PR TITLE
denoising_autoencoder: Change loss, increase epochs, improve visualization, use seed

### PR DESCRIPTION
**Pull request by ozabluda**
1. Change loss from mse to binary_crossentropy. This
- 1. trains slightly faster (stops improving after 30 epochs, as opposed to 40 for mse
- 2. has slightly lower **mse** test error 0.015480 vs 0.015491, when trained with mse loss.
- 3. most of the time produces better and/or less blurry result

2. Increase epochs from 10 to 30.
3. use seed. this allows comparison between mse and binary_crossentropy
4. Improve visualization (add orig images, increase amount) 

denoisingautoencoder bce2

@roatienza

### CODE REVIEW

**Comment by ozabluda** [comment]
Before (mse):
denoisingautoencoder mse

After (binary_crossentropy):
denoisingautoencoder bce

File: examples/mnist_denoising_autoencoder.py [hunk 1]
```diff
@@ -28,6 +28,9 @@
 from keras.datasets import mnist
 import numpy as np
 import matplotlib.pyplot as plt
+from PIL import Image
```
Comment (fchollet) [inline examples/mnist_denoising_autoencoder.py]: You could do that with Matplotlib, to avoid a new dependency

File: examples/mnist_denoising_autoencoder.py [hunk 2]
```diff
@@ -115,26 +118,32 @@
...
 # Predict the Autoencoder output from corrupted test images
 x_decoded = autoencoder.predict(x_test_noisy)
 
 # Display the 1st 8 corrupted and denoised images
-imgs = np.concatenate([x_test_noisy[:8], x_decoded[:8]])
-imgs = imgs.reshape((4, 4, image_size, image_size))
+rows, cols = 10, 30
+num = rows * cols
+imgs = np.concatenate([x_test[:num], x_test_noisy[:num], x_decoded[:num]])
+imgs = imgs.reshape((rows*3, cols, image_size, image_size))
```
Comment (fchollet) [inline examples/mnist_denoising_autoencoder.py]: Use PEP8 syntax throughout

File: examples/mnist_denoising_autoencoder.py [hunk 3]
```diff
@@ -115,26 +118,32 @@
...
+imgs = (imgs*255).astype(np.uint8)
 plt.figure()
 plt.axis('off')
-plt.title('Corrupted Input: top 2 rows, Output is Denoised Input: last 2 rows')
+plt.title('Original images: top rows, '
+          'Corrupted Input: middle rows, '
+          'Denoised Input:  third rows')
 plt.imshow(imgs, interpolation='none', cmap='gray')
-plt.savefig('corrupted_and_denoised.png')
+Image.fromarray(imgs).save("corrupted_and_denoised.png")
```
Comment (fchollet) [inline examples/mnist_denoising_autoencoder.py]: Use `'` as string delimiter for consistency

File: examples/mnist_denoising_autoencoder.py (hunk 1 above)
Comment (ozabluda) [reply examples/mnist_denoising_autoencoder.py]: Couldn't figure out how to remove all the whitespace around. PIL is used in the examples/ already.
https://user-images.githubusercontent.com/14060629/34853831-c931deec-f6ea-11e7-8d43-8093eb0a3705.png

File: examples/mnist_denoising_autoencoder.py (hunk 3 above)
Comment (ozabluda) [reply examples/mnist_denoising_autoencoder.py]: Oops, sorry. Used `"` and non-pep8 to mark my temporary hacks.

File: examples/mnist_denoising_autoencoder.py (hunk 1 above)
Comment (fchollet) [reply examples/mnist_denoising_autoencoder.py]: I understand that it's better to do the image stacking and formatting in Numpy, to avoid whitespace and so on. But you're only using PIL to convert your Numpy array to an image, and I'm saying you can use Matplotlib to do this step, e.g. `plt.imshow(imgs, cmap='gray')

Comment (ozabluda) [reply examples/mnist_denoising_autoencoder.py]: Using Matplotlib to save the image leaves whitespace around it. I couldn't figure out how to remove it.  `bbox_inches='tight', pad_inches=0, plt.tight_layout()` didn't work.

Compare matplotlib (whitespace)
https://user-images.githubusercontent.com/14060629/34853831-c931deec-f6ea-11e7-8d43-8093eb0a3705.png

to PIL (no whitespace)
https://user-images.githubusercontent.com/14060629/34853907-31da9830-f6eb-11e7-8da5-5756dee2610a.png
